kernel/private/rest/classes/ezpkernelrest.php

```diff
@@ -0,0 +1,294 @@
+<?php
+/**
+ * File containing the ezpKernelTreeMenu class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+class ezpKernelRest implements ezpKernelHandler
+{
+    /**
+     * Indicates is request has been properly initialized
+     *
+     * @var bool
+     */
+    protected $isInitialized = false;
+
+    /**
+     * Hash of internal settings
+     *
+     * @var array
+     */
+    protected $settings;
+
+    /**
+     * Current siteaccess data
+     *
+     * @var array
+     */
+    protected $access;
+
+    /**
+     * @var eZURI
+     */
+    protected $uri;
+
+    private $responseWriterClass;
+
+    private static $response;
+
+    public function __construct( array $settings = array(), $responseWriterClass )
```

Comment:
You should specify a default value for `$responseWriterClass` as it comes after an argument with default value.